child-logger.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+var defaultLevels = require('./default-levels.js');
+var makeLogMethod = require('./log-method');
+var errors = require('./errors');
+
+module.exports = ChildLogger;
+
+function ChildLogger(config) {
+    this.mainLogger = config.mainLogger;
+    this.path = config.path;
```

Comment:
Okay so I can see `createChild` being once-only, but then you need a `getChild` API which doesn't error, use cases like the following I'd consider to be idiomatic:
```
function doAThing(deps) {
  var myLogger = deps.logger.getChild('thing');
  ...
}
```
or even
```
function whatever(deps) {
  someLibraryCall({
    logger: deps.logger.getChild('someLibrary');
  });
}
```

Forcing application developers to cache and get "reference" handling as you call it right is too tall an ask for dubious gains; imo it even smells like classic "foolish consistency".

Also the primary point of such a feature, in my view, is context injection; how does this API support this?  E.g. a large part of the point is to DRY off contextual extra arguments, ideally we could do things like:
```
function foo(req, deps) {
  var logger = deps.logger.getChild('fooRequest', {
    extra: {
      userId: req.userId,
      // ... other context
    }
  });
}
```

However perhaps such context decoration runs at exact odds to any sort of child logger caching... in fact I'm largely confused as to why you're caching them at all on the parent logger now that I think through it more... what even are we solving for here again?